### PR TITLE
Cirrus: Detect unicode control chars. in source

### PR DESCRIPTION
**Pull request by cevich**
#### What this PR does / why we need it:

Use the python script provided as part of the RHSB-2021-007 advisory to
mitigate CVE-2021-42574 and CVE-2021-42694.  This could be automated
(overall) in a more elegant way, but for now running this on the podman
source (and vendored code) should be good enough.

#### How to verify it

The "Test Code Consistency" task will pass without error.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

**I did not write this python script**.  Also, I added configuration to ignore all `README` files because otherwise the script does find something to complain about:

```
./vendor/github.com/rivo/uniseg/README.md:17 unicode control characters: ['\u200d']
```

I'm hesitant to exclude all our docs, since there's "code" in some of them (examples).

### CODE REVIEW

**Comment by Luap99** [comment]
Should this script exclude the vendor directory? I mean we have no control about this anyway.

**Comment by cevich** [comment]
> I mean we have no control about this anyway.

I spent some time thinking about that, and decided that we should include it: We want to know right away if vendored-in code was compromised.  That's a much better situation than not knowing and/or assuming they've done their own checking. In certain special cases, we can add things into the exclude list as-needed.

File: contrib/cirrus/find_unicode_control2.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+By default the script takes one or more files or directories and looks for
+unicode control characters in all text files.  To narrow down the files, provide
+a config file with the -c command line, defining a scan_exclude list, which
+should be a list of regular expressions matching paths to exclude from the scan.
+
+There is a second mode enabled with -p which when set to 'all', prints all
+control characters and when set to 'bidi', prints only the 9 bidirectional
+control characters.
+"""
+from __future__ import print_function
```
Comment (jwhonce) [inline contrib/cirrus/find_unicode_control2.py:13]: You are invoking python3 on the shebang line so this import is not needed.

File: contrib/cirrus/find_unicode_control2.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+import importlib
+from stat import *
+
+def _unicode(line, encoding):
+    if isinstance(line, str):
+        return line
+    return line.decode(encoding)
+
+import platform
+if platform.python_version()[0] == '2':
```
Comment (jwhonce) [inline contrib/cirrus/find_unicode_control2.py:25]: Again, we know python version is >= 3 from shebang line.

**Review by jwhonce — COMMENTED** [review]
Nits on checking python versions < 3

File: contrib/cirrus/find_unicode_control2.py (hunk 2 above)
Comment (cevich) [reply contrib/cirrus/find_unicode_control2.py:25]: Not my code...and I wish there was a URL I could point at in a comment.  Sadly there isn't, this was downloaded directly from the advisory mentioned :cry:

**Comment by cevich** [comment]
Ugh, I give up.  It's way to fragile to expect 'no output' == 'success'  and this script isn't written in a way that seems easy to modify with a simple 0 or 1 exit code.

**Pull request closed without merging**